applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/adjoint_nodal_displacement_response_function.cpp

```diff
@@ -0,0 +1,228 @@
+
+
+namespace Kratos
+{
+    AdjointNodalDisplacementResponseFunction::AdjointNodalDisplacementResponseFunction(ModelPart& rModelPart, Parameters ResponseSettings)
+    : AdjointStructuralResponseFunction(rModelPart, ResponseSettings)
+    {
+        ModelPart& r_model_part = this->GetModelPart();
+        
+        // This response function currently only works in 3D!
+        ProcessInfo& r_current_process_info = r_model_part.GetProcessInfo();
+        const unsigned int domain_size =
+            static_cast<unsigned int>(r_current_process_info[DOMAIN_SIZE]);
+        KRATOS_ERROR_IF(domain_size != 3) << "Invalid DOMAIN_SIZE: " << domain_size << std::endl;
+
+        // Get id of node where a displacement should be traced
+        mIdOfTracedNode = ResponseSettings["traced_node_id"].GetInt();
+
+        // Get the corresponding dof to the displacement which should be traced
+        // by this response function e.g. DISPLACEMENT_X, ROTATION_X,...
+        mTracedDofLabel = ResponseSettings["traced_dof"].GetString();
+
+        // Get pointer to traced node
+        mpTracedNode = r_model_part.pGetNode(mIdOfTracedNode);
+
+        // Check if variable for traced dof is valid
+        if( !( KratosComponents< VariableComponent< VectorComponentAdaptor<array_1d<double, 3> > > >::Has(mTracedDofLabel)) )
+            KRATOS_ERROR << "Specified traced DOF is not available. Specified DOF: " << mTracedDofLabel << std::endl;
+        else
+        {
+            const VariableComponentType& rTRACED_DOF =
+                KratosComponents<VariableComponentType>::Get(mTracedDofLabel);
+            KRATOS_ERROR_IF_NOT( mpTracedNode->SolutionStepsDataHas(rTRACED_DOF) )
+                << "Specified DOF is not available at traced node." << std::endl;
+        }
+
+        // Check if variable for traced adjoint dof is valid
+        if( !(KratosComponents< VariableComponent< VectorComponentAdaptor<array_1d<double, 3> > > >::Has(std::string("ADJOINT_") + mTracedDofLabel)) )
+        {
+            KRATOS_ERROR << "Specified traced adjoint DOF is not available." << std::endl;
+        }
+        else
+        {
+            const VariableComponentType& rTRACED_ADJOINT_DOF =
+                KratosComponents<VariableComponentType>::Get(std::string("ADJOINT_") + mTracedDofLabel);
+            KRATOS_ERROR_IF_NOT( mpTracedNode->SolutionStepsDataHas(rTRACED_ADJOINT_DOF) )
+                << "Specified adjoint DOF is not available at traced node." << std::endl;
+        }
+
+        mDisplacementValue = 0.0;
+        this->GetNeighboringElementPointer();
+    }
+
+    // ==============================================================================
+    AdjointNodalDisplacementResponseFunction::~AdjointNodalDisplacementResponseFunction(){}
+
+    // ==============================================================================
+    void AdjointNodalDisplacementResponseFunction::GetNeighboringElementPointer()
```

Comment:
What are you using this function for?

Reply:
Within this function I am looking for one element which is bounded by the traced node. This element is necessary in order to assemble the adjoint load case into the system for the adjoint problem.